internal/gatewayapi/route.go

```diff
@@ -995,6 +996,9 @@ func (t *Translator) processTCPRouteParentRefs(tcpRoute *TCPRouteContext, resour
 				},
 				TLS: &ir.TLS{Terminate: irTLSConfigs(listener.tlsSecrets)},
 			}
+			if irListener.TLS.Terminate != nil {
```

Comment:
nice !
a comment in here saying enable http/1.1 & http/2 alpn protocols by default would help

Reply:
Added a comment.

Reply:
just realized this is the logic for TCPRoute, what about HTTPRoute and GRPCRoute ?

Reply:
rethinking about this logic, here's my sugession
1. dont parse/set ALPN here
2. in the ClientTrafficPolicy layer, if ALPN is set, then set it in the IR
3. If the xds layer, if ALPN is nil, set to http/1.1 and http/2, if set, set to the value in ALPN field

Reply:
Done.